File: Scala/abc188_b/Main.scala
import scala.util.chaining.*
import scala.io.StdIn.readLine

@main def Main(): Unit =
  val n = readInputInt()
  val a = readInputArray()
  val b = readInputArray()

  val result = solve(a, b)
  println(result)

def readInputInt(): Int =
  readLine().toInt

def readInputArray(): Array[Int] =
  readLine().split(" ").map(_.toInt)

def solve(a: Array[Int], b: Array[Int]): String =
  (a zip b)
    .map { case (x, y) => x * y }
    .sum
    .pipe:
      case 0 => "Yes"
      case _ => "No"

File: Ruby/abc188_b/main.rb
def get_numbers
  gets.split.map(&:to_i)
end

def get_input_as_string
  gets.chomp
end

def get_input_as_int
  gets.chomp.to_i
end

def solve(n, a, b)
  sum = 0
  for i in 0...n 
    sum += a[i] * b[i]
  end

  if sum == 0 
    "Yes"
  else
    "No"
  end
end

def main
  n = get_input_as_int
  a = get_numbers
  b = get_numbers

  puts solve(n, a, b)
end

main
